index.js

```diff
@@ -112,90 +130,107 @@ module.exports = {
             available: function() { return false; }
         };
 
-        var red = {};
-        for (i in RED) {
-            if (RED.hasOwnProperty(i) && !/^(init|start|stop)$/.test(i)) {
-                var propDescriptor = Object.getOwnPropertyDescriptor(RED,i);
-                Object.defineProperty(red,i,propDescriptor);
-            }
-        }
-
-        red["_"] = function(messageId) {
-            return messageId;
+        var red = {
+            _: v => v
         };
 
-        redNodes.init({events:events,settings:settings, storage:storage,log:log,});
-        redNodes.registerType("helper", helperNode);
+        Object.keys(this._RED).filter(prop => !/^(init|start|stop)$/.test(prop))
+        .forEach(prop => {
+            const propDescriptor = Object.getOwnPropertyDescriptor(this._RED, prop);
+            Object.defineProperty(red, prop, propDescriptor);
+        });
+
+        const redNodes = this._redNodes;
+        redNodes.init({
+            events: this._events,
+            settings: settings,
+            storage:storage,
+            log:this._log
+        });
+        redNodes.registerType("helper", function (n) {
+            redNodes.createNode(this, n);
+        });
+
         if (Array.isArray(testNode)) {
-            for (i = 0; i < testNode.length; i++) {
-                testNode[i](red);
-            }
+            testNode.forEach(fn => {
+                fn(red);
+            });
         } else {
             testNode(red);
         }
-        redNodes.loadFlows().then(function() {
-            redNodes.startFlows();
-            should.deepEqual(testFlow, redNodes.getFlows().flows);
-            cb();
-        });
-    },
 
-    unload: function() {
+        redNodes.loadFlows()
+            .then(() => {
+                redNodes.startFlows();
+                should.deepEqual(testFlow, redNodes.getFlows().flows);
+                cb();
+            });
+    }
+
+    unload() {
         // TODO: any other state to remove between tests?
-        redNodes.clearRegistry();
-        logSpy.restore();
-        // internal API
-        context.clean({allNodes:[]});
-        return redNodes.stopFlows();
-    },
+        this._redNodes.clearRegistry();
+        this._logSpy.restore();
+        this._sandbox.reset();
```

Comment:
we don't *restore* the sandbox, because that would kill all of the spies we created in our method proxies.  so we just reset the call history

Reply:
I suggest you _restore_ the sandbox here, assuming you move the spies you created to the load method.  You can then remove `this._logSpy.restore()` since they will all be restored.